src/transformStateWithClones.js

```diff
@@ -6,8 +6,36 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClones = [];
+  const recentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(recentState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        action.keysToRemove.forEach(key => {
+          delete recentState[key];
+        });
+        break;
+
+      case 'clear':
+        for (const key in recentState) {
+          delete recentState[key];
+        }
+        break;
+      default:
+        throw new Error();
```

Comment:
Usually it's a bad idea to throw without any context, because this means that consumer of this function (You in the different part of the code, or someone else in the team) has to know that this function is not safe and has to check if it will not throw.
I would change it to `console.error('Your message')`
You can disable linting for this line only.

Reply:
Also, if you are already throwing, always throw with some meaningful message like: 
```throw new Error(`Unknown action of type: ${action.type}`);```